sklearn/model_selection/_split.py

```diff
@@ -1481,6 +1481,50 @@ class GroupShuffleSplit(ShuffleSplit):
         If None, the random number generator is the RandomState instance used
         by `np.random`.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.model_selection import GroupShuffleSplit
+    >>> X = np.array([[1, 2], [3, 4], [5, 6]])
+    >>> y = np.array([1, 2, 1])
+    >>> groups = np.array([1, 2, 3, 4, 5])
+    >>> gss = GroupShuffleSplit()
```

Comment:
you need to always provide `random_state` where it's a parameter, to be sure it's deterministic.